### ISSUE TITLE
NuGet PackageReference should recommend PrivateAssets=all

### ISSUE THREAD

**Issue by menees** [Sergio0694/PolySharp#19]
### Description

PolySharp's NuGet [PackageReference recommendation](https://www.nuget.org/packages/PolySharp/#package-reference) is currently:

``` XML
<PackageReference Include="PolySharp" Version="1.7.1" />
```

Since PolySharp is a source-only polyfill provider, it should recommend a more restricted scope PackageReference like this:

``` XML
<PackageReference Include="PolySharp" Version="1.7.1">
  <PrivateAssets>all</PrivateAssets>
  <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
</PackageReference>
```

### Rationale

This prevents any package dependencies from flowing to parent projects as described at [Controlling dependency assets](https://learn.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#controlling-dependency-assets). It's also the recommendation of every other source-only polyfill project I'm familiar with (e.g., [PolyKit.Embedded](https://www.nuget.org/packages/PolyKit.Embedded#package-reference), [Polyfill.NET](https://www.nuget.org/packages/Polyfill.NET#package-reference), and [Nullable](https://www.nuget.org/packages/Nullable#package-reference)).

### Drawbacks

None that I'm aware of.

**Comment by Sergio0694** [comment]
Yeah that's a good point. I usually just recommend using `PrivateAssets="all"`, that alone should do the trick. I'm not aware of a way to change that suggestion on NuGet though, that's not something that I see an option for on the website 🤔

I guess I could at least add a section about this in the readme though to help folks.

**Comment by menees** [comment]
I believe you need to set the MSBuild property `DevelopmentDependency` to `true` as discussed at [DevelopmentDependency support for PackageReference](https://github.com/NuGet/Home/wiki/DevelopmentDependency-support-for-PackageReference). It can be in your .csproj files (e.g., [PolyKit.Embedded.csproj](https://github.com/Tenacom/PolyKit/search?q=developmentDependency), [Polyfill.NET.csproj](https://github.com/ssa3512/Polyfill.NET/search?q=developmentDependency)), or it can be in a .nuspec file (e.g., [Nullable.nuspec](https://github.com/manuelroemer/Nullable/search?q=developmentDependency)).

**Comment by kzu** [comment]
This uncovered an annoying inconsistency in nugetizer. I've now fixed it, published the fix, and set the PR to fix this :).
